crates/ere-zisk/src/lib.rs

```diff
@@ -0,0 +1,151 @@
+            stats: true,
+            ..Default::default()
+        };
+
+        // Execute
+        let mut emu = Emu::new(&rom);
+        emu.run(inputs, &options, None::<Box<dyn Fn(_)>>);
+
+        if !emu.terminated() {
+            return Err(ZiskError::Execute(ExecuteError::EmulationNotTerminate).into());
+        }
+
+        // FIXME: An ugly hack to get cycle count because `emu.ctx.stats`
+        //        doesn't have an public getter for it.
+        let total_num_cycles = emu
+            .ctx
+            .stats
+            .report()
+            .split_once("total steps = ")
+            .and_then(|(_, stats)| {
+                stats
+                    .split_whitespace()
+                    .next()
+                    .and_then(|steps| steps.parse::<u64>().ok())
+            })
+            .ok_or(ZiskError::Execute(ExecuteError::TotalStepsNotFound))?;
+
+        Ok(ProgramExecutionReport::new(total_num_cycles))
+    }
+
+    fn prove(&self, _: &Input) -> Result<(Vec<u8>, ProgramProvingReport), zkVMError> {
+        todo!()
+    }
+
+    fn verify(&self, _: &[u8]) -> Result<(), zkVMError> {
+        todo!()
+    }
+}
+
+#[cfg(test)]
+mod execute_tests {
+    use super::*;
+
+    fn get_compiled_test_zisk_elf() -> Result<PathBuf, ZiskError> {
+        let test_guest_path = get_execute_test_guest_program_path();
+        RV64_IMA_ZISK_ZKVM_ELF::compile(&test_guest_path)
+    }
+
+    fn get_execute_test_guest_program_path() -> PathBuf {
+        let workspace_dir = env!("CARGO_WORKSPACE_DIR");
+        PathBuf::from(workspace_dir)
+            .join("tests")
+            .join("zisk")
+            .join("execute")
+            .join("basic")
+            .canonicalize()
+            .expect("Failed to find or canonicalize test guest program at <CARGO_WORKSPACE_DIR>/tests/execute/zisk")
+    }
+
+    #[test]
+    fn test_execute_zisk_dummy_input() {
+        let elf_path = get_compiled_test_zisk_elf()
+            .expect("Failed to compile test ZisK guest for execution test");
+
+        let mut input_builder = Input::new();
+        let n: u32 = 42;
+        let a: u16 = 42;
+        input_builder.write(n);
+        input_builder.write(a);
+
+        let zkvm = EreZisk::new(elf_path);
+
+        let result = zkvm.execute(&input_builder);
+
+        if let Err(e) = &result {
+            panic!("Execution error: {:?}", e);
+        }
+    }
+
+    // FIXME: The emulator panics due to some memory reading issue, but ideally
+    //        it shouldn't panic but instead returning error indicating the
+    //        non-zero exit code.
+    #[test]
+    #[should_panic]
+    fn test_execute_zisk_no_input_for_guest_expecting_input() {
+        let elf_path = get_compiled_test_zisk_elf()
+            .expect("Failed to compile test ZisK guest for execution test");
+
+        let empty_input = Input::new();
+
+        let zkvm = EreZisk::new(elf_path);
+        let _ = zkvm.execute(&empty_input);
+    }
```

Comment:
If the guest program panics or calls `std::process::exit` explicitly, `zisk`'s emulator will also panic with some memory reading issue, and the guest program's panic message is printed to stdout so we couldn't catch it easily, so here I use `#[should_panic]` for now but ideally we should check the execution error to be something like `HaltWithNonZeroExitCode`.